### DISCUSSION TITLE
How can I create a custom map for testing instead of a procedure generated?

### DISCUSSION THREAD

**Discussion by viniciusferrao** [diasurgical/DevilutionX#6272] [Q&A] [answered]
Considering I want to make some changes on the code, how can I effectively test them?

For example. I want to change the effects of Cauldron in the game, but to achieve that I must have a map with a fixed Cauldron on it, is there a way to force it? Use a custom loaded map? If I want a Shrine can I force it to be a specific one?

Thanks.

**Comment by AJenbo** [comment, marked as answer]
https://github.com/diasurgical/modding-tools/blob/master/Tiled/README.md

**Reply by AJenbo** [reply]
you can load it with the `map <name>` chat command in game.

**Reply by AJenbo** [reply]
... ps. if you have issue running the graphics tool then please try this beta: https://github.com/diasurgical/d1-graphics-tool/suites/13546502032/artifacts/745237809

**Comment by StephenCWills** [comment]
This is not an answer to your question about custom maps, but you might be interested to know that you can grab information about the dungeon seed on a given level with `/seedinfo` and then regenerate the level using the `restart` command in a debug build. So if you find a cauldron or shrine, you can save the seed for that level and replay it.

https://github.com/diasurgical/devilutionX/wiki/Debug-Commands

If you simply need to test changes to the way these objects work, it may be even easier. For example, put a direct call to `OperateShrineMysterious(*MyPlayer)` in the handler for `DebugToggle` and then press that key to invoke it.

https://github.com/diasurgical/devilutionX/blob/ad7020be20e6f59b22ed0d05d3b3efd40a37cc3e/Source/diablo.cpp#L1834-L1841
